src/frontend/src/flows/authenticate/index.ts

```diff
@@ -322,11 +255,79 @@ async function handleAuthSuccess(
   return {
     userNumber: loginResult.userNumber,
     connection: loginResult.connection,
-    sendDelegationMessage: () =>
-      authContext.postMessageCallback({
-        kind: "authorize-client-success",
-        delegations: [parsed_signed_delegation],
-        userPublicKey: Uint8Array.from(userKey),
-      }),
+    parsedSignedDelegation: parsed_signed_delegation,
+    userKey,
   };
 }
+
+/** Run the authentication flow, including postMessage protocol, offering to authenticate
+ * using an existing anchor or creating a new anchor, etc.
+ */
+export const authenticationFlow = async (
+  connection: Connection
+): Promise<void> => {
+  const container = document.getElementById("pageContent") as HTMLElement;
+  render(html`<h1>starting authentication</h1>`, container);
+  const result = await authenticationProtocol({
+    authenticate: async (authContext) => {
+      const authSuccess = await authenticatePage(connection, authContext);
+      await recoveryWizard(authSuccess.userNumber, authSuccess.connection);
+      return authSuccess;
+    },
+    onInvalidOrigin: (result) =>
+      displayError({
+        title: "Invalid Derivation Origin",
+        message: `"${result.authContext.authRequest.derivationOrigin}" is not a valid derivation origin for "${result.authContext.requestOrigin}"`,
+        detail: result.message,
+        primaryButton: "Continue",
+      }),
+
+    onProgress: (status) => {
+      switch (status) {
+        case "waiting":
+          render(
+            html`<h1>waiting for authentication data from dapp...</h1>`,
+            container
+          );
+          break;
+        case "validating":
+          render(html`<h1>validating authentication data...</h1>`, container);
+          break;
+        default:
+          unreachable(status);
+          break;
+      }
+    },
+  });
+
+  switch (result) {
+    case "orphan":
+      await displayError({
+        title: "Invalid Data",
+        message: `It looks like you were sent here for authentication, but no dapp requested authentication.`,
+        primaryButton: "Home",
+      });
+
+      location.hash = "";
+      window.location.reload();
+      break;
+    case "failure":
+      render(
+        html`<h1>
+          Something went wrong during authentication. Authenticating dapp was
+          notified and you may close this page.
+        </h1>`,
+        container
+      );
+      break;
+    case "success":
+      render(
+        html`<h1>Authentication sucessful. You may close this page.</h1>`,
```

Comment:
Would it make sense to have screenshots of those pages too?

Reply:
These will almost never be seen, hence the very simplistic html... I think it's not worth the extra code for extracting the pages + adding them to the showcase